src/main/java/in/org/projecteka/hiu/dataprocessor/DataAvailabilityListener.java

```diff
@@ -46,11 +46,9 @@ public void subscribe()  {
             DataAvailableMessage dataAvailableMessage = deserializeMessage(message);
             logger.info(String.format("Received notification of data availability for transaction id : %s", dataAvailableMessage.getTransactionId()));
             logger.info(String.format("Processing data from file : %s", dataAvailableMessage.getPathToFile()));
-            try {
-                new HealthDataProcessor(healthDataRepository, dataFlowRepository, new Decryptor()).process(dataAvailableMessage);
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+            HealthDataProcessor healthDataProcessor = new HealthDataProcessor(healthDataRepository, dataFlowRepository, new Decryptor());
+            healthDataProcessor.registerHITypeResourceHandler(new DiagnosticReportResourceProcessor());
```

Comment:
Why can't this also be constructor parameter?

Reply:
There would be potentially many such processors.